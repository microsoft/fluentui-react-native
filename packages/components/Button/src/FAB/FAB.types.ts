import { ButtonSlotProps, ButtonCoreTokens, ButtonCoreProps } from '../Button.types';
import { ShadowProps } from '@fluentui-react-native/experimental-shadow';

export const fabName = 'FAB';

export interface FABSlotProps extends ButtonSlotProps {
  shadow?: ShadowProps;
}

export interface FABProps extends ButtonCoreProps {
  showContent?: boolean;
}

export interface FABType {
  props: FABProps;
  tokens: ButtonCoreTokens;
  slotProps: FABSlotProps;
}
