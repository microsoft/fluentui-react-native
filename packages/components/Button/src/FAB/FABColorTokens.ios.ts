import type { Theme } from '@fluentui-react-native/framework';
import type { TokenSettings } from '@fluentui-react-native/use-styling';

import type { FABTokens } from './FAB.types';

export const defaultFABColorTokens: TokenSettings<FABTokens, Theme> = (t: Theme): FABTokens => ({
  // Coloring same as primary
  backgroundColor: t.colors.brandedBackground,
  color: t.colors.brandedContent,
  borderColor: t.colors.brandedBorder,
  iconColor: t.colors.brandedIcon,
  disabled: {
    backgroundColor: t.colors.brandedDisabledBackground,
    color: t.colors.brandedDisabledContent,
    borderColor: t.colors.brandedDisabledBorder,
    iconColor: t.colors.brandedDisabledIcon,
  },
  pressed: {
    backgroundColor: t.colors.brandedPressedBackground,
    color: t.colors.brandedPressedContent,
    borderColor: t.colors.brandedPressedBorder,
    iconColor: t.colors.brandedPressedIcon,
  },
  focused: {
    backgroundColor: t.colors.brandedFocusedBackground,
    color: t.colors.brandedFocusedContent,
    borderColor: t.colors.brandedFocusedBorder,
    iconColor: t.colors.brandedFocusedIcon,
  },
});
