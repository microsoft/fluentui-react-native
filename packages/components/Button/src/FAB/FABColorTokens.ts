import { Theme } from '@fluentui-react-native/framework';
import { TokenSettings } from '@fluentui-react-native/use-styling';
import { ButtonCoreTokens } from '../Button.types';

export const defaultFABColorTokens: TokenSettings<ButtonCoreTokens, Theme> = (t: Theme): ButtonCoreTokens => ({
  // coloring same as primary
  backgroundColor: t.colors.brandedBackground,
  color: t.colors.brandedContent,
  borderColor: t.colors.brandedBorder,
  iconColor: t.colors.brandedIcon,
  disabled: {
    backgroundColor: t.colors.brandedDisabledBackground,
    color: t.colors.brandedDisabledContent,
    borderColor: t.colors.brandedDisabledBorder,
    iconColor: t.colors.brandedDisabledIcon,
  },
  hovered: {
    backgroundColor: t.colors.brandedHoveredBackground,
    color: t.colors.brandedHoveredContent,
    borderColor: t.colors.brandedHoveredBorder,
    iconColor: t.colors.brandedHoveredIcon,
  },
  pressed: {
    backgroundColor: t.colors.brandedPressedBackground,
    color: t.colors.brandedPressedContent,
    borderColor: t.colors.brandedPressedBorder,
    iconColor: t.colors.brandedPressedIcon,
  },
  focused: {
    backgroundColor: t.colors.brandedFocusedBackground,
    color: t.colors.brandedFocusedContent,
    borderColor: t.colors.brandedFocusedBorder,
    iconColor: t.colors.brandedFocusedIcon,
  },
});
