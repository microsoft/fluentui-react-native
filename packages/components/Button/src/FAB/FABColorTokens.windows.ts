import { Theme } from '@fluentui-react-native/framework';
import { TokenSettings } from '@fluentui-react-native/use-styling';
import { ButtonCoreTokens } from '../Button.types';
import { shadowStyleFromGlobalToken } from '../shadowStyle';

export const defaultFABColorTokens: TokenSettings<ButtonCoreTokens, Theme> = (t: Theme): ButtonCoreTokens => ({
  ...shadowStyleFromGlobalToken(8),
  // coloring same as primary
  backgroundColor: t.colors.brandBackground,
  color: t.colors.neutralForegroundOnBrand,
  borderColor: t.colors.brandStroke1,
  iconColor: t.colors.neutralForegroundOnBrand,
  disabled: {
    backgroundColor: t.colors.neutralBackgroundDisabled,
    color: t.colors.neutralForegroundDisabled,
    borderColor: t.colors.neutralStrokeDisabled,
    iconColor: t.colors.neutralForegroundDisabled,
  },
  hovered: {
    backgroundColor: t.colors.brandBackgroundHover,
    color: t.colors.neutralForegroundOnBrandHover,
    borderColor: t.colors.brandBackgroundHover,
    iconColor: t.colors.neutralForegroundOnBrandHover,
  },
  pressed: {
    backgroundColor: t.colors.brandBackgroundPressed,
    color: t.colors.neutralForegroundOnBrandPressed,
    borderColor: t.colors.brandBackgroundPressed,
    iconColor: t.colors.neutralForegroundOnBrandPressed,
  },
  focused: {
    backgroundColor: t.colors.brandBackgroundHover,
    color: t.colors.neutralForegroundOnBrandHover,
    borderColor: t.colors.strokeFocus2,
    iconColor: t.colors.neutralForegroundOnBrandHover,
  },
});
