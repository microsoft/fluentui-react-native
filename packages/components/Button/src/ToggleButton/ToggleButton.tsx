/** @jsx withSlots */
import * as React from 'react';
import { Pressable } from 'react-native';
import { ActivityIndicator } from '@fluentui-react-native/experimental-activity-indicator';
import { ToggleButtonProps, toggleButtonName, ToggleButtonType } from './ToggleButton.types';
import { TextV1 as Text } from '@fluentui-react-native/text';
import { stylingSettings } from './ToggleButton.styling';
import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
import { Icon } from '@fluentui-react-native/icon';
import { createIconProps } from '@fluentui-react-native/interactive-hooks';
import { buttonLookup } from '../Button';
import { useToggleButton } from './useToggleButton';

export const ToggleButton = compose<ToggleButtonType>({
  displayName: toggleButtonName,
  ...stylingSettings,
  slots: {
    root: Pressable,
    icon: Icon,
    content: Text,
  },
  useRender: (userProps: ToggleButtonProps, useSlots: UseSlots<ToggleButtonType>) => {
    const iconProps = createIconProps(userProps.icon);
    const toggleButton = useToggleButton(userProps);

    // grab the styled slots
    const Slots = useSlots(userProps, (layer) => buttonLookup(layer, toggleButton.state, userProps));

    // now return the handler for finishing render
    return (final: ToggleButtonProps, ...children: React.ReactNode[]) => {
      const { icon, iconPosition, iconOnly, loading, accessibilityLabel, ...mergedProps } = mergeProps(toggleButton.props, final);
      const shouldShowIcon = !loading && icon;

      if (__DEV__ && iconOnly) {
        React.Children.forEach(children, (child) => {
          if (typeof child === 'string') {
            console.warn('iconOnly should not be set when Button has content.');
          }
        });
      }

      let childText = '';
      if (accessibilityLabel === undefined) {
        React.Children.forEach(children, (child) => {
          if (typeof child === 'string') {
            childText = child; // We only automatically support the one child string.
          }
        });
      }
      const label = accessibilityLabel ?? childText;

      return (
        <Slots.root {...mergedProps} accessibilityLabel={label}>
          {loading && <ActivityIndicator />}
          {shouldShowIcon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
          {React.Children.map(children, (child) =>
            typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
          )}
          {shouldShowIcon && iconPosition === 'after' && <Slots.icon {...iconProps} />}
        </Slots.root>
      );
    };
  },
});
