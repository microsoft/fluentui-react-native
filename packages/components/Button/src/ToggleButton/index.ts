export * from './ToggleButton.types';
export * from './ToggleButton';
