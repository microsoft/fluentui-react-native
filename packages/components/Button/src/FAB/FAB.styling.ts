import { ButtonCoreTokens, ButtonSlotProps, ButtonCoreProps } from '../Button.types';
import { Theme, UseStylingOptions, buildProps } from '@fluentui-react-native/framework';
import { borderStyles, layoutStyles, fontStyles, shadowStyles } from '@fluentui-react-native/tokens';
import { buttonCoreStates } from '../Button.styling';
import { getTextMarginAdjustment } from '@fluentui-react-native/styling-utils';
import { defaultFABTokens } from './FABTokens';
import { defaultFABColorTokens } from './FABColorTokens';

export const stylingSettings: UseStylingOptions<ButtonCoreProps, ButtonSlotProps, ButtonCoreTokens> = {
  tokens: [defaultFABTokens, defaultFABColorTokens],
  states: [...buttonCoreStates],
  slotProps: {
    root: buildProps(
      (tokens: ButtonCoreTokens, theme: Theme) => ({
        style: {
          display: 'flex',
          alignItems: 'center',
          flexDirection: 'row',
          alignSelf: 'flex-start',
          justifyContent: 'center',
          width: tokens.width,
          backgroundColor: tokens.backgroundColor,
          ...borderStyles.from(tokens, theme),
          ...layoutStyles.from(tokens, theme),
          ...shadowStyles.from(tokens, theme),
        },
        elevation: tokens.elevation,
      }),
      ['backgroundColor', 'width', ...borderStyles.keys, ...layoutStyles.keys],
    ),
    content: buildProps(
      (tokens: ButtonCoreTokens, theme: Theme) => ({
        style: {
          color: tokens.color,
          ...getTextMarginAdjustment(),
          ...(tokens.spacingIconContentBefore && { marginStart: tokens.spacingIconContentBefore }),
          ...fontStyles.from(tokens, theme),
        },
      }),
      ['color', 'spacingIconContentBefore', ...fontStyles.keys],
    ),
    icon: buildProps(
      (tokens: ButtonCoreTokens) => ({
        style: {
          tintColor: tokens.iconColor,
        },
        height: tokens.iconSize,
        width: tokens.iconSize,
      }),
      ['iconColor', 'iconSize'],
    ),
    shadow: buildProps(
      (tokens: ButtonCoreTokens) => ({
        shadowToken: tokens.shadowToken,
      }),
      ['shadowToken'],
    ),
  },
};
