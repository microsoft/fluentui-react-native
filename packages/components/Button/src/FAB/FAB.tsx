/** @jsx withSlots */
import * as React from 'react';
import { View } from 'react-native';
import { fabName, FABProps, FABType } from './FAB.types';
import { TextV1 as Text } from '@fluentui-react-native/text';
import { compose, UseSlots } from '@fluentui-react-native/framework';
import { Icon } from '@fluentui-react-native/icon';

export const FAB = compose<FABType>({
  displayName: fabName,
  slots: {
    root: View,
    icon: Icon,
    content: Text,
  },
  useRender: (_userProps: FABProps, _useSlots: UseSlots<FABType>) => {
    return (_final: FABProps, ..._children: React.ReactNode[]) => {
      return null; // Only implemented for mobile endpoints
    };
  },
});
