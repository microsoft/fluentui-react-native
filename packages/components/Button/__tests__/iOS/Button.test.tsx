import { buttonLegacyTest } from '../ButtonLegacy.test';
import { buttonV1Tests } from '../ButtonV1.test';
import { fabTests } from '../FAB.test';
import { toggleButtonTests } from '../ToggleButton.test';

buttonLegacyTest();
buttonV1Tests();
fabTests();
toggleButtonTests();
