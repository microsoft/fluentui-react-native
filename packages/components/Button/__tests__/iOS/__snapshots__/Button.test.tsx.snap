// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests Button circular 1`] = `
<View
  accessibilityLabel="Circular Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="circular"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 9999,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Circular Button
  </Text>
</View>
`;

exports[`Button component tests Button composed 1`] = `
<View
  accessibilityLabel="Composed Button with RNText"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "dynamicTypeRamp": undefined,
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": undefined,
        "lineHeight": undefined,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
        "maximumFontSize": undefined,
      }
    }
  >
    Composed Button with RNText
  </Text>
</View>
`;

exports[`Button component tests Button customized 1`] = `
<View
  accessibilityLabel="Custom Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Custom Button
  </Text>
</View>
`;

exports[`Button component tests Button default 1`] = `
<View
  accessibilityLabel="Default Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Default Button
  </Text>
</View>
`;

exports[`Button component tests Button disabled 1`] = `
<View
  accessibilityLabel="Disabled Button"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#f3f2f1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#d2d0ce",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Disabled Button
  </Text>
</View>
`;

exports[`Button component tests Button large 1`] = `
<View
  accessibilityLabel="Large Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 11,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Large Button
  </Text>
</View>
`;

exports[`Button component tests Button primary 1`] = `
<View
  accessibilityLabel="Primary Button"
  accessibilityRole="button"
  accessible={true}
  appearance="primary"
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Primary Button
  </Text>
</View>
`;

exports[`Button component tests Button small 1`] = `
<View
  accessibilityLabel="Small Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="small"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 64,
      "padding": 7,
      "paddingHorizontal": 11,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Small Button
  </Text>
</View>
`;

exports[`Button component tests Button square 1`] = `
<View
  accessibilityLabel="Square Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="square"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 0,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Square Button
  </Text>
</View>
`;

exports[`Button component tests Button subtle 1`] = `
<View
  accessibilityLabel="Subtle Button"
  accessibilityRole="button"
  accessible={true}
  appearance="subtle"
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#323130",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Subtle Button
  </Text>
</View>
`;

exports[`Button default 1`] = `
<View
  accessibilityLabel="Default Button"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 24,
          "minWidth": 32,
          "paddingEnd": 6,
          "paddingStart": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#323130",
            "fontFamily": "System",
            "fontSize": 13,
            "fontWeight": "400",
            "margin": 0,
            "marginEnd": 2,
            "marginStart": 2,
          }
        }
      >
        Default Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`Custom FAB with no shadow(iOS) 1`] = `
<View
  accessibilityLabel="Custom FAB with no shadow(iOS)"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  iconPosition="before"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 9999,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 56,
      "minWidth": 56,
      "padding": 20,
      "width": undefined,
    }
  }
>
  <Text
    accessible={true}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Arial",
        "fontSize": 16,
        "fontWeight": "400",
        "margin": 0,
      }
    }
  >
    ♣
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 8,
        "marginTop": 0,
      }
    }
  >
    Custom FAB with no shadow(iOS)
  </Text>
</View>
`;

exports[`Default FAB (iOS) 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 9999,
      "bottom": undefined,
      "display": "flex",
      "end": undefined,
      "justifyContent": "center",
      "left": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginEnd": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginStart": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "minHeight": 56,
      "minWidth": 56,
      "right": undefined,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.12,
      "shadowRadius": 1,
      "start": undefined,
      "top": undefined,
      "width": undefined,
    }
  }
>
  <View
    accessibilityLabel="Default FAB (iOS)"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    enableFocusRing={true}
    focusable={true}
    iconPosition="before"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#0f6cbd",
        "borderBottomWidth": undefined,
        "borderColor": "#0f548c",
        "borderEndWidth": undefined,
        "borderLeftWidth": undefined,
        "borderRadius": 9999,
        "borderRightWidth": undefined,
        "borderStartWidth": undefined,
        "borderTopWidth": undefined,
        "borderWidth": undefined,
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": undefined,
        "justifyContent": "center",
        "minHeight": 56,
        "minWidth": 56,
        "padding": 20,
        "paddingBottom": undefined,
        "paddingEnd": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingStart": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.14,
        "shadowRadius": 4,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={true}
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Arial",
          "fontSize": 16,
          "fontWeight": "400",
          "margin": 0,
        }
      }
    >
      ♣
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={0}
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "System",
          "fontSize": 13,
          "fontWeight": "400",
          "margin": 0,
          "marginBottom": 0,
          "marginEnd": 0,
          "marginStart": 8,
          "marginTop": 0,
        }
      }
    >
      Default FAB (iOS)
    </Text>
  </View>
</View>
`;

exports[`ToggleButton default 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Default Button"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "checked": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  enableFocusRing={true}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0f6cbd",
      "borderColor": "#0f548c",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 8,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginEnd": 0,
        "marginStart": 0,
        "marginTop": 0,
      }
    }
  >
    Default Button
  </Text>
</View>
`;
